### PR TITLE
Add sourcemap, catalog_assets and parse to OptionBuilder

### LINKED ISSUE #808: OptionsBuilder: support for sourcemap
From https://asciidoctor.org/docs/user-manual/#ruby-api-options

> `:sourcemap`
> Keeps track of the file and line number for each parsed block. (Useful for tooling applications where the association between the converted output and the source file is important). 
> default value: `false`
> Allowed values: boolean

Unless it is set to `true`, `StructuralNode#getSourceLocation()` returns `null`.

---
There is no support in `OptionsBuilder`.

Is it just missing (in this case I can create a PR to add it), or is there a reason why it is not present?

---
Workaround:
```java
OptionsBuilder.options()
                .option("sourcemap", true)
```

### PR DESCRIPTION
**Pull request by jmini**
Fixes #808

This PR do not contains the version bump to `2.1.0`

### CODE REVIEW

File: asciidoctorj-api/src/main/java/org/asciidoctor/Options.java [hunk 1]
```diff
@@ -124,10 +127,30 @@ public void setSafe(SafeMode safeMode) {
         this.options.put(SAFE, safeMode.getLevel());
     }
 
+    /**
+     * Keeps track of the file and line number for each parsed block. (Useful for tooling applications where the association between the converted output and the source file is important).
+     * 
+     * @param sourcemap
+     *            value.
+     */
+    public void setSourcemap(boolean sourcemap) {
```
Comment (robertpanzer) [inline asciidoctorj-api/src/main/java/org/asciidoctor/Options.java]: Would you mind updating this test to use this setter?
https://github.com/asciidoctor/asciidoctorj/blob/1a33db5703859b5c8ce9aa840270350f65f8e263/asciidoctorj-core/src/test/java/org/asciidoctor/WhenAsciiDocIsRenderedToDocument.java#L263-L268

**Comment by robertpanzer** [comment]
Could you also please add an entry to the change log? :)

**Comment by jmini** [comment]
I implemented the changes.

**Comment by robertpanzer** [comment]
Thank you! 
Sorry for the delay. I am not as fast in reviewing as I’d like to be.

**Comment by jmini** [comment]
Thank you for merging this. No worry about the delay. 
The pace of reviews in the Asciidoctor(J) project is great.

**Pull request merged**